### PR TITLE
Most accurate team

### PR DESCRIPTION
**Pull request by sciencefixion**
I'm throwing up this PR for a little help. My most_accurate_team method passes my test with our default setup, but fails the spec harness test. I'm not sure why, but I'm inclined to think it may be a math issue. Let me know if you have any thoughts on this. I'm having the same issue with worst_coach. I skipped both tests even though they pass with my tests. @evilaspaas1 @gabichuelas @rrabinovitch

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -315,9 +315,57 @@ def worst_coach(season)
     coach_losses.max_by { |coach| losses_by_coach[coach] }
```
Comment (gabichuelas) [inline lib/stat_tracker.rb:311]: @sciencefixion this is the only part of `worst_coach` that doesn't make sense to me so far... I think you were trying to say 
```
loss_count = losses_by_coach.max_by { |coach, count| count }
```
and then you can return the coach in question by calling 
`losses_by_coach.key(loss_count)`

Comment (gabichuelas) [reply lib/stat_tracker.rb:311]: `hash.key(value)` returns the `key` that corresponds to given `value`

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -315,9 +315,57 @@ def worst_coach(season)
...
+
+    season_goals_by_team = season_games.inject(Hash.new(0)) do |season_team_goals, game|
+      season_team_goals[game.team_id] += game.goals.to_i
+      season_team_goals
+    end
+
+    season_team_accuracy = {}
+    season_goals_by_team.each do |team_goals_id, goals|
+      season_shots_by_team.each do |team_shots_id, shots|
+        if team_goals_id == team_shots_id
```
Comment (gabichuelas) [inline lib/stat_tracker.rb]: @sciencefixion i find this conditional statement problematic because your code below only runs if the # of goals == # of shots, but that wouldn't be the case, right? We'd have more shots than goals? So maybe you don't need a conditional at all, you just want to do the math that you wrote below, I think.

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -315,9 +315,57 @@ def worst_coach(season)
...
+    # possible helper method
+    season_games_by_team = season_games.group_by do |game|
+    game.team_id
+    end
+
+    season_team_ids = [season_games_by_team.keys].flatten
+
+    season_teams = teams.find_all { |team| season_team_ids.include?(team.team_id) }
+
+    season_teams_by_id = season_teams.group_by do |team|
```
Comment (gabichuelas) [inline lib/stat_tracker.rb]: If `season_teams` is an array of `Team` objects, each of them should have a different `team_id`, so I'm not super sure why you're turning them into a hash, as each pair will just represent one team..?

Comment (gabichuelas) [reply lib/stat_tracker.rb]: I think maybe it makes more sense to iterate over `season_teams` with something like this (?) :
```
season_teams.find { |team| team.team_id == most_accurate_team_id }
```
and then call team_name on the result of that ^^

**Review by gabichuelas — CHANGES_REQUESTED** [review]
@sciencefixion I left some comments, I hope they make sense!!!

File: lib/stat_tracker.rb (hunk 2 above)
Comment (sciencefixion) [reply lib/stat_tracker.rb]: My variables could use better naming I think. I am trying take my hashes of season_team_goals and season_team_shots and produce a new hash that has the accuracy percentage. I obviously struggled here because I have nested iteration. But I'm trying to match the keys so my math is working on the right numbers and my result is still connected to the team_id.

TL;DR The conditional is there to match the two keys in separate hashes. The result should be a new has with team_id => accuracy percentage.

**Comment by sciencefixion** [comment]
This method now passes all tests including spec harness.
@evilaspaas1 @rrabinovitch @gabichuelas

**Review by araspaas — APPROVED** [review]
if they are all now passing then looks good to me.

**Review by gabichuelas — APPROVED** [review]
